sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridPublisherAsyncClient.java

```diff
@@ -57,26 +76,130 @@
 
         this.hostname = hostname;
         this.eventDataSerializer = eventDataSerializer;
+        this.eventClass = eventClass;
     }
 
     /**
-     * Get the service version of the Rest API.
-     * @return the Service version of the rest API
+     * Generate a shared access signature to provide time-limited authentication for requests to the Event Grid
+     * service with the latest Event Grid service API defined in {@link EventGridServiceVersion#getLatest()}.
+     * @param endpoint the endpoint of the Event Grid topic or domain.
+     * @param expirationTime the time in which the signature should expire, no longer providing authentication.
+     * @param keyCredential  the access key obtained from the Event Grid topic or domain.
+     *
+     * @return the shared access signature string which can be used to construct an instance of
+     * {@link AzureSasCredential}.
      */
-    public EventGridServiceVersion getServiceVersion() {
-        return this.serviceVersion;
+    public static String generateSas(String endpoint, AzureKeyCredential keyCredential, OffsetDateTime expirationTime) {
+        return generateSas(endpoint, keyCredential, expirationTime, EventGridServiceVersion.getLatest());
+    }
+
+    /**
+     * Generate a shared access signature to provide time-limited authentication for requests to the Event Grid
+     * service.
+     * @param endpoint the endpoint of the Event Grid topic or domain.
+     * @param expirationTime the time in which the signature should expire, no longer providing authentication.
+     * @param keyCredential  the access key obtained from the Event Grid topic or domain.
+     * @param apiVersion the EventGrid service api version defined in {@link EventGridServiceVersion}
+     *
+     * @return the shared access signature string which can be used to construct an instance of
+     * {@link AzureSasCredential}.
+     */
+    public static String generateSas(String endpoint, AzureKeyCredential keyCredential, OffsetDateTime expirationTime,
+        EventGridServiceVersion apiVersion) {
+        try {
+            String resKey = "r";
+            String expKey = "e";
+            String signKey = "s";
+
+            Charset charset = StandardCharsets.UTF_8;
+            endpoint = endpoint + "?api-version=" + apiVersion.getVersion();
+            String encodedResource = URLEncoder.encode(endpoint, charset.name());
+            String encodedExpiration = URLEncoder.encode(expirationTime.atZoneSameInstant(ZoneOffset.UTC).format(
+                DateTimeFormatter.ofPattern("M/d/yyyy h:m:s a")),
```

Comment:
This could be defined as static, so we do not create a new instance everytime.
DateTimeFormatter formatter = DateTimeFormatter.ofPattern("M/d/yyyy h:m:s a");

Reply:
Changed to use static variable.